clientd3d/dialog.c

```diff
@@ -663,6 +674,41 @@ void DisplayDescription(object_node *obj, BYTE flags, char *description,
 }
 /************************************************************************/
 /*
+* GetRaritySuffix:  Provide a rarity suffix based on the given rarity type.
+*/
+std::string GetRaritySuffix(int rarity) 
+{
+	std::string suffix;
+	
+	switch(rarity)
+	{
+        case ITEM_RARITY_GRADE_NORMAL:
+            suffix = "";
+			break;
+        case ITEM_RARITY_GRADE_MAGIC:
+            return GetString(hInst, IDS_RARITY_GRADE_MAGIC);
+			break;
+        case ITEM_RARITY_GRADE_RARE:
+            return GetString(hInst, IDS_RARITY_GRADE_RARE);
+			break;
+        case ITEM_RARITY_GRADE_LEGENDARY:
+            return GetString(hInst, IDS_RARITY_GRADE_LEGENDARY);
+			break;
+        case ITEM_RARITY_GRADE_MYSTERIOUS:
+            return GetString(hInst, IDS_RARITY_GRADE_MYSTERIOUS);
+			break;
+        case ITEM_RARITY_GRADE_CURSED:
+            return GetString(hInst, IDS_RARITY_GRADE_CURSED);
+			break;
+		default:
+			debug(("Unknown rarity grade\n"));
```

Comment:
Log the value of the unknown rarity, too.

Reply:
Added rarity value to unknown rarity grade debug message as requested.